iroha/src/sumeragi.rs

```diff
@@ -412,7 +415,14 @@ impl Sumeragi {
             .clear();
         self.transactions_awaiting_receipts.write().await.clear();
         let previous_role = self.network_topology.role(&self.peer_id);
-        self.network_topology.shift_peers_by_one();
+        if self.number_of_view_changes < self.n_topology_shifts_before_reshuffle {
```

Comment:
May be would be convenient to create separate method with such if in order to avoid repetition?

Reply:
The code and intent is a bit different though, in sumeragi it applies shifts by one, and here all together.